pmd-cataloger/src/main/java/sfdc/sfdx/scanner/pmd/FileExaminer.java

```diff
@@ -0,0 +1,110 @@
+package sfdc.sfdx.scanner.pmd;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.jar.JarEntry;
+import java.util.jar.JarInputStream;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/***
+ * Examines files and helps scout for jar files and XMLs
+ * @author rmohan
+ * Feb 2020
+ */
+public class FileExaminer {
+
+    enum FileType {
+        JAR(".jar"),
+        XML(".xml");
+
+        String suffix;
+
+        FileType(String suffix) {
+            this.suffix = suffix;
+        }
+    }
+
+
+    /***
+     * Find XML in a given path
+     * @param pathString - could be a directory with classes and/or jar files, or a single jar file
+     * @return a list of XML files found
+     */
+    public List<String> findXmlInPath(String pathString) {
```

Comment:
Changing the name to `findXmlFilesInPath` would make it clear that this expects to return multiple files. Also, this method needs more inline commenting so it's clear exactly what it's doing and why.

Reply:
I can modify the method name. 
Also, if you feel the code doesn't look intuitive enough, I can add more inline commenting. I've been trying to keep up readability of code so that it doesn't need a lot of inline commenting. Also, as the code grows, these comments can turn into unintentional misdirection.

http://nicolecarpenter.github.io/2016/03/08/clean-code-chapter-4-comments.html

Reply:
The two schools of thought are "good code should be self-documenting" and "even good code requires documentation". I agree with the second. The code can't tell me anything other than what it's _currently_ doing. The comments can tell me what the code is _supposed_ to do, which is very helpful for debugging. The best coding advice I've ever received was "never assume the person debugging your code next month understands it, especially if you know that person will be you."